kong/utils_test.go

```diff
@@ -567,6 +570,363 @@ func TestFillUpstreamsDefaults(T *testing.T) {
+							HTTPStatuses: []int{200, 302},
+							Interval:     Int(0),
+							Successes:    Int(0),
+						},
+						HTTPPath:               String("/"),
+						HTTPSVerifyCertificate: Bool(true),
+						Type:                   String("http"),
+						Timeout:                Int(1),
+						Unhealthy: &Unhealthy{
+							HTTPFailures: Int(0),
+							HTTPStatuses: []int{
+								429, 404,
+								500, 501, 502, 503, 504, 505,
+							},
+							TCPFailures: Int(0),
+							Timeouts:    Int(0),
+							Interval:    Int(0),
+						},
+					},
+					Passive: &PassiveHealthcheck{
+						Healthy: &Healthy{
+							HTTPStatuses: []int{
+								200, 201, 202, 203, 204, 205, 206, 207, 208, 226,
+								300, 301, 302, 303, 304, 305, 306, 307, 308,
+							},
+							Successes: Int(0),
+						},
+						Type: String("http"),
+						Unhealthy: &Unhealthy{
+							HTTPFailures: Int(0),
+							HTTPStatuses: []int{429, 500, 503},
+							TCPFailures:  Int(0),
+							Timeouts:     Int(0),
+						},
+					},
+				},
+				HashOn:           String("none"),
+				HashFallback:     String("none"),
+				HashOnCookiePath: String("/"),
+			},
+		},
+		{
+			name: "fills defaults for all fields except algorithm and hash_on, leaves name unchanged",
+			upstream: &Upstream{
+				Name:      String("upstream1"),
+				Algorithm: String("consistent-hashing"),
+				HashOn:    String("ip"),
+			},
+			expected: &Upstream{
+				Name:      String("upstream1"),
+				Algorithm: String("consistent-hashing"),
+				Slots:     Int(10000),
+				Healthchecks: &Healthcheck{
+					Active: &ActiveHealthcheck{
+						Concurrency: Int(10),
+						Healthy: &Healthy{
+							HTTPStatuses: []int{200, 302},
+							Interval:     Int(0),
+							Successes:    Int(0),
+						},
+						HTTPPath:               String("/"),
+						HTTPSVerifyCertificate: Bool(true),
+						Type:                   String("http"),
+						Timeout:                Int(1),
+						Unhealthy: &Unhealthy{
+							HTTPFailures: Int(0),
+							HTTPStatuses: []int{
+								429, 404,
+								500, 501, 502, 503, 504, 505,
+							},
+							TCPFailures: Int(0),
+							Timeouts:    Int(0),
+							Interval:    Int(0),
+						},
+					},
+					Passive: &PassiveHealthcheck{
+						Healthy: &Healthy{
+							HTTPStatuses: []int{
+								200, 201, 202, 203, 204, 205, 206, 207, 208, 226,
+								300, 301, 302, 303, 304, 305, 306, 307, 308,
+							},
+							Successes: Int(0),
+						},
+						Type: String("http"),
+						Unhealthy: &Unhealthy{
+							HTTPFailures: Int(0),
+							HTTPStatuses: []int{429, 500, 503},
+							TCPFailures:  Int(0),
+							Timeouts:     Int(0),
+						},
+					},
+				},
+				HashOn:           String("ip"),
+				HashFallback:     String("none"),
+				HashOnCookiePath: String("/"),
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			u := tc.upstream
+			if err := FillEntityDefaults(u, schema); err != nil {
+				t.Errorf(err.Error())
+			}
```

Comment:
```suggestion
			require.NoError(t, FillEntityDefaults(u, schema))
```

Simpler and more succint?

Reply:
I like simpler :D